app/src/main/java/com/arkbuilders/arkdrop/presentation/feature/qrcodescanner/QRCodeScannerViewModel.kt

```diff
@@ -0,0 +1,39 @@
+package com.arkbuilders.arkdrop.presentation.feature.qrcodescanner
+
+import android.content.Intent
+import android.graphics.Rect
+import android.net.Uri
+import android.view.MotionEvent
+import android.view.View
+import com.google.mlkit.vision.barcode.common.Barcode
+
+class QRCodeScannerViewModel(barcode: Barcode) {
+
+    var boundingRect: Rect = barcode.boundingBox!!
+    var qrContent: String = ""
+    var qrCodeTouchCallback = { v: View, e: MotionEvent -> false } //no-op
+
+    init {
+        when (barcode.valueType) {
+            Barcode.TYPE_URL -> {
+                qrContent = barcode.url!!.url!!
```

Comment:
Similarly to https://github.com/ARK-Builders/ARK-Drop/pull/1/files#r1586184421, could we also handle `null` data more elegantly here instead of non-null assertion?